src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -120,6 +124,18 @@ private void OnStopRude(object obj)
                 IntPtr.Zero);
         }
 
+        private void OnStopRuder(object obj)
```

Comment:
Libuv reference counts and uses that to determine if it can shutdown loop. `OnStop` unreferences thread' `OnStopRude` calls `Dispose` on all the items still referenced and they call close which hopefully unreferences them. So `OnStopRuder` explicitly unreferences them.